variables.tf

```
@@ -0,0 +1,110 @@
+  description = "The Zone to launch the DB instance. MongoDB sharding instance does not support multiple-zone. If it is a multi-zone and vswitch_id is specified, the vswitch must in one of them. "
+  default     = ""
+}
+
+variable "vswitch_id" {
+  description = "The virtual switch ID to launch DB instances in one VPC. "
+  default     = ""
+}
+
+variable "account_password" {
+  description = "Password of the root account. It is a string of 6 to 32 characters and is composed of letters, numbers, and underlines. "
+  default     = ""
+}
+
+variable "security_ip_list" {
+  description = "List of IP addresses allowed to access all databases of an instance. The list contains up to 1,000 IP addresses, separated by commas. Supported formats include 0.0.0.0/0, 10.23.12.24 (IP), and 10.23.12.24/24 (Classless Inter-Domain Routing (CIDR) mode. /24 represents the length of the prefix in an IP address. The range of the prefix length is [1,32]). System default to [`127.0.0.1`]. "
+  type        = list(string)
+  default     = []
+}
+
+
+variable "mongo_list" {
+  description = "The mongo-node count can be purchased is in range of [2, 32]. "
+  type        = list(map(string))
+  default     = []
+}
+
+variable "shard_list" {
+  description = "The shard-node count can be purchased is in range of [2, 32]. "
+  type        = list(map(string))
+  default     = []
+}
+
+variable "backup_period" {
+  description = "MongoDB sharding Instance backup period. It is required when backup_time was existed. Valid values: [Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday]. Default to [Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday] "
+  type        = list(any)
+  default     = []
+}
+
+variable "backup_time" {
+  description = "MongoDB sharding instance backup time. It is required when backup_period was existed. In the format of HH:mmZ- HH:mmZ. Time setting interval is one hour. Default to a random time, like `23:00Z-24:00Z`. "
+  default     = "23:00Z-24:00Z"
```

"Default to a random time " 跟 `"23:00Z-24:00Z"` 不是一回事吧